conftest.py

```diff
@@ -36,14 +36,32 @@ def empty_netcdf4_file(tmp_path: Path) -> str:
     return str(filepath)
 
 
+@pytest.fixture
+def netcdf4_file_with_scale(tmp_path: Path) -> str:
+    filepath = tmp_path / "air.nc"
+
+    # Set up example xarray dataset
+    with xr.tutorial.open_dataset("air_temperature") as ds:
+        # Save it to disk as netCDF (in temporary directory)
+        ds.to_netcdf(
+            filepath,
+            format="NETCDF4",
+            encoding={"air": {"dtype": "float32", "scale_factor": 0.01}},
+        )
+
+    return str(filepath)
+
+
 @pytest.fixture
 def netcdf4_file(tmp_path: Path) -> str:
     filepath = tmp_path / "air.nc"
 
     # Set up example xarray dataset
     with xr.tutorial.open_dataset("air_temperature") as ds:
         # Save it to disk as netCDF (in temporary directory)
-        ds.to_netcdf(filepath, format="NETCDF4")
+        air_encoding = ds["air"].encoding
+        air_encoding["_FillValue"] = -9999
```

Comment:
This fixes a warning `SerializationWarning: saving variable air with floating point data as an integer dtype without any _FillValue to use for NaNs`